OpenRA.Mods.Common/Traits/AffectsShroud.cs

```diff
@@ -34,7 +34,8 @@ public abstract class AffectsShroud : ConditionalTrait<AffectsShroudInfo>, ITick
 		readonly HashSet<PPos> footprint;
 
 		[Sync] CPos cachedLocation;
-		[Sync] bool cachedTraitDisabled;
+		[Sync] WDist cachedRange;
+		[Sync] public bool CachedTraitDisabled { get; private set; }
```

Comment:
Please make this `protected` instead of `public`.  It only needs to be visible to subclasses.